program/src/balance.rs

```diff
@@ -31,28 +31,31 @@ pub fn get_target_balance(
     );
 
     let total_delegated_lamports: token::Result<Lamports> = validators
-        .iter_entries()
+        .iter_active()
         .map(|v| v.stake_accounts_balance)
         .sum();
 
     let total_lamports = total_delegated_lamports.and_then(|t| t + undelegated_lamports)?;
 
-    let total_weights: u64 = validators.iter_entries().map(|v| v.weight.0 as u64).sum();
-
     // We only want to target validators that are not in the process of being
-    // removed. For now, those are all the validators. Once we add validator
-    // removal, we need to take the removal flag into account here.
-    let num_active_validators = validators.len() as u64;
+    // removed.
+    let num_active_validators = validators.iter_active().count() as u64;
 
-    // Target a weighted distribution. A division by zero means that
-    // `total_weights` is zero, that means there are no active validators.
+    // No active validators.
+    if num_active_validators == 0 {
+        return Err(LidoError::EmptySetOfValidators);
```

Comment:
The error message here might be misleading now, because the set of validators can be non-empty and we still return this error. If the error is not used anywhere else, we can rename the error.